File: events/src/main/java/com/vzk/events/models/Event.java
package com.vzk.events.models;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.UUID;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity(name = "events")
public class Event {
    @Id
    @Column(name = "id")
    private UUID id;

    @Basic
    @Column(name = "name")
    private String name;

    @Basic
    @Column(name = "info")
    private String info;

    @Basic
    @Column(name = "date")
    private String date;

    @Basic
    @Column(name = "is_active")
    private boolean isActive;

    @Basic
    @Column(name = "game")
    private UUID game;

    @Basic
    @Column(name = "enemy_logo")
    private String enemyLogo;
}

File: ui/vzk-ui/src/app/models/event/event.ts
export class Event {
  id: string;
  name: string;
  game: string;
  date: Date;
  info: string;
  enemyLogo: string;

  constructor(id: string, title: string, datetime: Date, description: string, game: string, enemyURL: string) {
    this.id = id;
    this.name = title;
    this.date = datetime;
    this.info = description;
    this.game = game;
    this.enemyLogo = enemyURL;
  }
}
